Roles/Impostor/DollMaster.cs

```diff
@@ -0,0 +1,319 @@
+using AmongUs.GameOptions;
+using TOHE.Roles.Neutral;
+using UnityEngine;
+using static TOHE.Options;
+using static TOHE.Translator;
+
+namespace TOHE.Roles.Impostor;
+
+internal class DollMaster : RoleBase
+{
+    //===========================SETUP================================\\
+    private const int Id = 28500;
+    private static readonly HashSet<byte> PlayerIds = [];
+    public static bool HasEnabled => PlayerIds.Any();
+    public override bool IsEnable => HasEnabled;
+    public override CustomRoles ThisRoleBase => CustomRoles.Shapeshifter;
+    public override Custom_RoleType ThisRoleType => Custom_RoleType.ImpostorKilling;
+    //==================================================================\\
+    private static readonly HashSet<byte> ReducedVisionPlayers = [];
+    public static bool isEnabled = false;
+    private static bool IsControllingPlayer = false;
+    private static bool ResetPlayerSpeed = false;
+    private static bool WaitToUnPossess = false;
+    private static PlayerControl controllingTarget = null; // Personal possessed player identifier for reference.
+    private static PlayerControl DollMasterTarget = null; // Personal possessed player identifier for reference.
+    private static float originalSpeed = 0;
+    private static Vector2 controllingTargetPos = new(0, 0);
+    private static Vector2 DollMasterPos = new(0, 0);
+
+    private static OptionItem DefaultKillCooldown;
+    private static OptionItem ShapeshiftCooldown;
+    private static OptionItem ShapeshiftDuration;
+    public static OptionItem CanKillAsMainBody;
+    public static OptionItem TargetDiesAfterPossession;
+
+    public override void SetupCustomOption()
+    {
+        SetupRoleOptions(Id, TabGroup.ImpostorRoles, CustomRoles.DollMaster);
+        DefaultKillCooldown = FloatOptionItem.Create(Id + 10, "KillCooldown", new(0f, 180f, 2.5f), 25f, TabGroup.ImpostorRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.DollMaster])
+            .SetValueFormat(OptionFormat.Seconds);
+        ShapeshiftCooldown = FloatOptionItem.Create(Id + 11, "DollMasterPossessionCooldown", new(0f, 180f, 2.5f), 25f, TabGroup.ImpostorRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.DollMaster])
+            .SetValueFormat(OptionFormat.Seconds);
+        ShapeshiftDuration = FloatOptionItem.Create(Id + 12, "DollMasterPossessionDuration", new(0f, 180f, 2.5f), 10f, TabGroup.ImpostorRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.DollMaster])
+            .SetValueFormat(OptionFormat.Seconds);
+        CanKillAsMainBody = BooleanOptionItem.Create(Id + 13, "DollMasterCanKillAsMainBody", true, TabGroup.ImpostorRoles, false).SetParent(Options.CustomRoleSpawnChances[CustomRoles.DollMaster]);
+        TargetDiesAfterPossession = BooleanOptionItem.Create(Id + 14, "DollMasterTargetDiesAfterPossession", false, TabGroup.ImpostorRoles, false).SetParent(Options.CustomRoleSpawnChances[CustomRoles.DollMaster]);

```

Comment:
that is just doppelganger lol

Reply:
Should I just remove the setting then? I wanted to add the setting so if someone makes it where you can't kill as the main body, there's an alternative option.

Reply:
yeah. and I mean it's really up to the host to make the settings fair, but not making a second doppel..